### PR TITLE
Update API AMI

### PR DESCRIPTION
**Pull request by davidsmejia**
## Issue Number

#315 

## Purpose/Implementation Notes

- Uses the latest ubuntu 22.04 AMI
- Adds docker install to user-data
- fixes tabs in user-data
- updates apt-get -> apt

## Types of changes

- Bugfix (non-breaking change which fixes an issue)

## Functional tests

n/a

## Checklist

- [X] Lint and unit tests pass locally with my changes

## Screenshots

N/A

### CODE REVIEW

File: infrastructure/api-configuration/api-server-instance-user-data.tpl.sh [hunk 1]
```diff
@@ -16,55 +16,61 @@ cd /home/ubuntu
...
+        sudo zip -r ../letsencryptdir.zip "../$(basename "$PWD")"
+
+        # And then cleanup the extra copy.
+        rm /etc/letsencrypt/nginx.conf
+
+        cd - || exit
+        mv /etc/letsencryptdir.zip .
+        aws s3 cp letsencryptdir.zip "s3://${scpca_portal_cert_bucket}/"
+        rm letsencryptdir.zip
+        else
```
Comment (arkid15r) [inline infrastructure/api-configuration/api-server-instance-user-data.tpl.sh:68]: mis-indentation?

File: infrastructure/api-configuration/api-server-instance-user-data.tpl.sh [hunk 2]
```diff
@@ -16,55 +16,61 @@ cd /home/ubuntu
...
+        zip_filename=$(aws s3 ls "${scpca_portal_cert_bucket}" | head -1 | awk '{print $4}')
+        aws s3 cp "s3://${scpca_portal_cert_bucket}/$zip_filename" letsencryptdir.zip
+        unzip letsencryptdir.zip -d /etc/
```
Comment (arkid15r) [inline infrastructure/api-configuration/api-server-instance-user-data.tpl.sh:71]: It seems that `letsencryptdir.zip` deserves its own constant declaration too.

File: infrastructure/api-configuration/api-server-instance-user-data.tpl.sh [hunk 3]
```diff
@@ -16,55 +16,61 @@ cd /home/ubuntu
...
+        # g3w4k4t5n3s7p7v8@alexslemonade.slack.com is the email address we
+        # have configured to forward mail to the #teamcontact channel in
+        # slack. Certbot will use it for "important account
+        # notifications".
+
+        # The certbot challenge cannot be completed until the aws_lb_target_group_attachment resources are created.
+        sleep 180
+        BASE_URL="scpca.alexslemonade.org"
+        if [[ ${stage} == "staging" ]]; then
+                certbot --nginx -d api.staging.$BASE_URL -n --agree-tos --redirect -m g3w4k4t5n3s7p7v8@alexslemonade.slack.com
```
Comment (arkid15r) [inline infrastructure/api-configuration/api-server-instance-user-data.tpl.sh:50]: May be a good idea to declare NOTIFICATION_EMAIL="g3w4k4t5n3s7p7v8@alexslemonade.slack.com"

**Review by arkid15r — APPROVED** [review]

**Comment by nozomione** [comment]
@davidsmejia  Besides the question regarding the `gnupg` package installation I asked you yesterday(and thank you for the answer!), the commands look good to me. Since I've never done this installation workflow, please skip my 'Approval' at this time and merge the code as soon as it's ready. Thank you, David!

File: infrastructure/api-configuration/api-server-instance-user-data.tpl.sh (hunk 1 above)
Comment (davidsmejia) [reply infrastructure/api-configuration/api-server-instance-user-data.tpl.sh]: good catch

**Comment by davidsmejia** [comment]
I like the idea for the improvements with adding more constants, I actually want to revisit this next term and improve the script overall. I think we can clean it up significantly and also make the logic more portable for other instances we want to spin up.

**Pull request merged**